src/server/scripts/Pet/pet_dk.cpp

```diff
@@ -139,7 +157,263 @@ class npc_pet_dk_ebon_gargoyle : public CreatureScript
+                // Prevent early victim engage
+                creature->SetReactState(REACT_PASSIVE);
+                _engageTimer = 0;
+            }
+
+            void IsSummonedBy(Unit* summoner) override
+            {
+                DoCast(summoner, SPELL_COPY_WEAPON, true);
+                DoCast(summoner, SPELL_DK_RUNE_WEAPON_MARK, true);
+                DoCast(me, SPELL_DK_DANCING_RUNE_WEAPON_VISUAL, true);
+                DoCast(me, SPELL_FAKE_AGGRO_RADIUS_8_YARD, true);
+                DoCast(me, SPELL_DK_RUNE_WEAPON_SCALING_01, true);
+                DoCast(me, SPELL_DK_RUNE_WEAPON_SCALING, true);
+                DoCast(me, SPELL_PET_SCALING__MASTER_SPELL_06__SPELL_HIT_EXPERTISE_SPELL_PENETRATION, true);
+                DoCast(me, SPELL_DK_PET_SCALING_03, true);
+
+                _events.ScheduleEvent(EVENT_SPELL_CAST_2, 6 * IN_MILLISECONDS);
+                _engageTimer = 1 * IN_MILLISECONDS;
+
+                me->SetRedirectThreat(summoner->GetGUID(), 100);
+            }
+
+            void MoveInLineOfSight(Unit* /*who*/) override { }
+            void AttackStart(Unit* /*who*/) override { }
+
+            void SetGUID(ObjectGuid guid, int32 type) override
+            {
+                switch (type)
+                {
+                    case DATA_INITIAL_TARGET_GUID:
+                        _targetGUID = guid;
+                        me->SetReactState(REACT_AGGRESSIVE);
+                        if (Unit* target = ObjectAccessor::GetUnit(*me, _targetGUID))
+                            Attack(target);
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (me->HasReactState(REACT_PASSIVE))
+                {
+                    if (_engageTimer <= diff)
+                        me->SetReactState(REACT_AGGRESSIVE);
+                    else
+                    {
+                        _engageTimer -= diff;
+                        return;
+                    }
+                }
+
+                if (me->IsInCombat() && (!me->GetVictim() || !me->IsValidAttackTarget(me->GetVictim())))
+                    EnterEvadeMode(EVADE_REASON_NO_HOSTILES);
+
+                if (!me->IsInCombat())
+                {
+                    Unit* ownerTarget = nullptr;
+                    if (Player* owner = me->GetCharmerOrOwner()->ToPlayer())
+                        ownerTarget = owner->GetSelectedUnit();
+
+                    // recognize which victim will be choosen
+                    if (ownerTarget && ownerTarget->GetTypeId() == TYPEID_PLAYER)
+                    {
+                        if (!ownerTarget->HasBreakableByDamageCrowdControlAura(ownerTarget))
+                            Attack(ownerTarget);
+                    }
+                    else if (ownerTarget && (ownerTarget->GetTypeId() != TYPEID_PLAYER) && IsInThreatList(ownerTarget))
```

Comment:
because the code above is logically different from the code of the PR, read both closely and you'll see how they differ.